[Platform] Linux build failure from missing pthread header

### Reproduction steps / Feature

```bash
1. `scripts/run_in_build_env.sh "scripts/build/build_examples.py --target linux-x64-all-clusters-no-ble build"`

2. 


2022-09-22 16:50:57 INFO    FAILED: obj/third_party/connectedhomeip/examples/platform/linux/app-main.NamedPipeCommands.cpp.o 
2022-09-22 16:50:57 INFO    g++ -MMD -MF obj/third_party/connectedhomeip/examples/platform/linux/app-main.NamedPipeCommands.cpp.o.d -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -Wall -Werror -Wextra -Wshadow -Wunreachable-code -Wvla -Wformat -Wformat-nonliteral -Wformat-security -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wno-cast-function-type -fdiagnostics-color -fno-strict-aliasing -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -Wno-implicit-fallthrough -std=gnu++14 -fno-rtti -Wnon-virtual-dtor -DCHIP_HAVE_CONFIG_H=1 -DCHIP_ADDRESS_RESOLVE_IMPL_INCLUDE_HEADER=\<lib/address_resolve/AddressResolve_DefaultImpl.h\> -DCHIP_MINMDNS_USE_EPHEMERAL_UNICAST_PORT=1 -DCHIP_MINMDNS_HIGH_VERBOSITY=0 -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/providers -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/src/include -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/src -Igen/include -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/zzz_generated/app-common -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/all-clusters-app/linux/include -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/config/standalone -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/third_party/nlassert/repo/include -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/third_party/nlio/repo/include -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/third_party/inipp/repo/inipp -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/zzz_generated -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/common/tracing -I../../examples/all-clusters-app/linux/third_party/connectedhomeip/third_party/jsoncpp/repo/include -c ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp -o obj/third_party/connectedhomeip/examples/platform/linux/app-main.NamedPipeCommands.cpp.o
2022-09-22 16:50:57 INFO    In file included from ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:22:
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp: In member function ‘CHIP_ERROR NamedPipeCommands::Start(std::string&, NamedPipeCommandDelegate*)’:
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:41:9: error: ‘pthread_create’ was not declared in this scope; did you mean ‘pthread_t’?
2022-09-22 16:50:57 INFO       41 |         pthread_create(&mChipEventCommandListener, nullptr, EventCommandListenerTask, reinterpret_cast<void *>(this)) == 0,
2022-09-22 16:50:57 INFO          |         ^~~~~~~~~~~~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/src/lib/support/CodeUtils.h:308:15: note: in definition of macro ‘VerifyOrReturnValue’
2022-09-22 16:50:57 INFO      308 |         if (!(expr))                                                                                                               \
2022-09-22 16:50:57 INFO          |               ^~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:40:5: note: in expansion of macro ‘VerifyOrReturnError’
2022-09-22 16:50:57 INFO       40 |     VerifyOrReturnError(
2022-09-22 16:50:57 INFO          |     ^~~~~~~~~~~~~~~~~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp: In member function ‘CHIP_ERROR NamedPipeCommands::Stop()’:
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:54:25: error: ‘pthread_cancel’ was not declared in this scope; did you mean ‘pthread_once_t’?
2022-09-22 16:50:57 INFO       54 |     VerifyOrReturnError(pthread_cancel(mChipEventCommandListener) == 0, CHIP_ERROR_CANCELLED);
2022-09-22 16:50:57 INFO          |                         ^~~~~~~~~~~~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/src/lib/support/CodeUtils.h:308:15: note: in definition of macro ‘VerifyOrReturnValue’
2022-09-22 16:50:57 INFO      308 |         if (!(expr))                                                                                                               \
2022-09-22 16:50:57 INFO          |               ^~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:54:5: note: in expansion of macro ‘VerifyOrReturnError’
2022-09-22 16:50:57 INFO       54 |     VerifyOrReturnError(pthread_cancel(mChipEventCommandListener) == 0, CHIP_ERROR_CANCELLED);
2022-09-22 16:50:57 INFO          |     ^~~~~~~~~~~~~~~~~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:57:25: error: ‘pthread_join’ was not declared in this scope; did you mean ‘pthread_t’?
2022-09-22 16:50:57 INFO       57 |     VerifyOrReturnError(pthread_join(mChipEventCommandListener, nullptr) == 0, CHIP_ERROR_SHUT_DOWN);
2022-09-22 16:50:57 INFO          |                         ^~~~~~~~~~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/src/lib/support/CodeUtils.h:308:15: note: in definition of macro ‘VerifyOrReturnValue’
2022-09-22 16:50:57 INFO      308 |         if (!(expr))                                                                                                               \
2022-09-22 16:50:57 INFO          |               ^~~~
2022-09-22 16:50:57 INFO    ../../examples/all-clusters-app/linux/third_party/connectedhomeip/examples/platform/linux/NamedPipeCommands.cpp:57:5: note: in expansion of macro ‘VerifyOrReturnError’
2022-09-22 16:50:57 INFO       57 |     VerifyOrReturnError(pthread_join(mChipEventCommandListener, nullptr) == 0, CHIP_ERROR_SHUT_DOWN);
2022-09-22 16:50:57 INFO          |     ^~~~~~~~~~~~~~~~~~~
2022-09-22 16:50:57 INFO    At global scope:
2022-09-22 16:50:57 INFO    cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
2022-09-22 16:50:57 INFO    ninja: build stopped: subcommand failed.
```
```


### Platform

other

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

--- a/examples/platform/linux/NamedPipeCommands.cpp
+++ b/examples/platform/linux/NamedPipeCommands.cpp
@@ -20,6 +20,7 @@
 
 #include <fcntl.h>
 #include <lib/support/CodeUtils.h>
+#include <pthread.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>

